contracts/Core/BlockManager.sol

```diff
@@ -173,27 +174,38 @@ contract BlockManager is Initializable, ACL, BlockStorage {
         }
     }
 
-    function confirmBlock() external initialized onlyRole(parameters.getBlockConfirmerHash()) {
+    function claimBlockReward() public initialized checkState(parameters.confirm()){
```

Comment:
this function should be `external` right? since its only called externally?
Also if the function is called externally then the following calculation should be done offchain. no need to do it here.
the proposed block id should be parameter of this function.

Reply:
We still need to verify onchain whether the block ID that is being sent is the first valid block. Otherwise, essentially anyone can confirm their block if it is in the proposal list.